pkg/entities/ie.go

```diff
@@ -620,3 +620,94 @@ func encodeInfoElementValueToBuff(element InfoElementWithValue, buffer []byte, i
 	}
 	return nil
 }
+
+// appendInfoElementValueToBuffer appends the encoded element value to the provided buffer.
+func appendInfoElementValueToBuffer(element InfoElementWithValue, buffer []byte) ([]byte, error) {
+	switch element.GetDataType() {
+	case OctetArray:
+		v := element.GetOctetArrayValue()
+		ieLen := element.GetInfoElement().Len
+		if ieLen < VariableLength {
+			// fixed length case
+			if len(v) != int(ieLen) {
+				return nil, fmt.Errorf("invalid value for fixed-length octet array: length mismatch")
+			}
+			buffer = append(buffer, v...)
+		} else if len(v) < 255 {
+			buffer = append(buffer, byte(len(v)))
```

Comment:
Forgive me if it is a dumb question. Why do we need to append the length or 255 before the element value?

Reply:
No worries. For variable-length elements, the encoding is as follows:
- if the length is < 255, write the length to the first byte, then write the binary data
- if the length is >= 255 and < 65,536, write 255 to the first byte, write the actual length to the next 2 bytes (uint16), then write the binary data
- if the length is >= 65,536, it is not supported

This is just IPFIX encoding. 255 is a sentinel value for the first byte.

Reply:
Got it. I was confused with the usage of ieLen and found it just a number to define whether it is a fixed-length or variable-length case/

Reply:
Yes, in the fixed-length case, `ieLen` matches the actual length. In the variable-length case, it is set to 65,535.